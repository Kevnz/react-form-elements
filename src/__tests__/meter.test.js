import React from 'react'
import { render, cleanup } from 'react-testing-library'

import Meter from '../meter'

describe('Meter component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(
        <Meter label="My Meter Element" initialValue={50} />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Meter Element')
      expect(label.textContent).toBe('My Meter Element')
    })
  })
  describe('Element attributes', () => {
    it('should have a value set', () => {
      const { container } = render(
        <Meter label="My Example Meter" initialValue={50} max={80} />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = container.querySelector('meter')

      expect(input.getAttribute('value')).toBe('50')
      expect(input.getAttribute('max')).toBe('80')
    })
  })
})
