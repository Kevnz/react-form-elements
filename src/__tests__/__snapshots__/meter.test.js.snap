// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meter component Element attributes should have a value set 1`] = `
<div
  class="rfe-form__row rfe-meter"
>
  <label
    for="id-2-1"
  >
    My Example Meter
  </label>
  <meter
    high="80"
    id="id-2-1"
    low="20"
    max="80"
    min="0"
    name="Progress"
    optimum="60"
    value="50"
  >
    at 
    50
    /
    80
  </meter>
</div>
`;

exports[`Meter component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row rfe-meter"
>
  <label
    for="id-1-1"
  >
    My Meter Element
  </label>
  <meter
    high="80"
    id="id-1-1"
    low="20"
    max="100"
    min="0"
    name="Progress"
    optimum="60"
    value="50"
  >
    at 
    50
    /
    100
  </meter>
</div>
`;
